src/schema/mbs_types.ts

```diff
@@ -0,0 +1,267 @@
+  info,
+) => {
+  const { scmurl } = parentValue;
+  const { instance } = args;
+  const nameWithCommit = _.last(_.split(scmurl, 'modules/'));
+  const [repoName, sha1] = _.split(nameWithCommit, '?#');
+  log('Getting commit object for %s:%s', repoName, sha1);
+  if (!repoName || !sha1) {
+    return {};
+  }
+  return await delegateToSchema({
+    schema: schema,
+    operation: 'query',
+    fieldName: 'distgit_commit',
+    args: {
+      commit_sha1: sha1,
+      instance,
+      namespace: 'modules',
+      repo_name: repoName,
+    },
+    context,
+    info,
+  });
+};
+
+const tagHistoryResolver: graphql.GraphQLFieldResolver<any, {}, any> = async (
+  parentValue,
+  args,
+  schemaContext,
+  info,
+) => {
+  const { context, name, stream, version } = parentValue;
+  const { instance } = args;
+  const nsvc = `${name}-${stream}-${version}.${context}`;
+  log('Delegating Koji tagging history query for NSVC %s', nsvc);
+  return await delegateToSchema({
+    schema,
+    operation: 'query',
+    fieldName: 'koji_build_history_by_nvr',
+    args: {
+      instance,
+      nvr: nsvc,
+    },
+    context: schemaContext,
+    info,
+  });
+};
+
+const tagsResolver: graphql.GraphQLFieldResolver<any, {}, any> = async (
+  parentValue,
+  args,
+  schemaContext,
+  info,
+) => {
+  const { context, name, stream, version } = parentValue;
+  const { instance } = args;
+  const nsvc = `${name}-${stream}-${version}.${context}`;
+  log('Delegating Koji tags query for NSVC %s', nsvc);
+  return await delegateToSchema({
+    schema: schema,
+    operation: 'query',
+    fieldName: 'koji_build_tags_by_nvr',
+    args: {
+      instance,
+      nvr: nsvc,
+    },
+    context: schemaContext,
+    info,
+  });
+};
+
+export const MbsBuildType = new GraphQLObjectType<MbsBuildFields, {}>({
+  name: 'MbsBuildType',
+  fields: {
+    commit: {
+      description:
+        'Commit object correponding to the source Git commit of the build',
+      type: CommitObject,
+      args: {
+        instance: {
+          type: DistGitInstanceInputType,
+          description: 'Dist-git instance identifer',
+          defaultValue: 'fp',
+        },
+      },
+      resolve: commitResolver,
+    },
+    context: {
+      description: 'Module context identifier. The ‘C’ in NSVC',
+      type: GraphQLNonNull(GraphQLString),
+    },
+    id: {
+      description: 'Module build ID',
+      type: GraphQLNonNull(GraphQLInt),
+    },
+    koji_tag: {
+      description:
+        'The corresponding Koji tag where the module components are built',
+      type: GraphQLNonNull(GraphQLString),
+    },
+    name: {
+      description: 'Name of the package. The ‘N’ in NSVC.',
+      type: GraphQLNonNull(GraphQLString),
+    },
+    owner: {
+      description: 'Owner/initiator of the module build',
+      type: GraphQLNonNull(GraphQLString),
+    },
+    scmurl: {
+      description: 'URL of the Git commit from which the module is built',
+      type: GraphQLNonNull(GraphQLString),
```

Comment:
It is necessary to check this. If `GraphQLNonNull` is really necessary here. Ordinary scratch builds can be built without source url. It could be the same for MBS builds.

Reply:
Yeah, I found some builds without `scmurl`. Fixed.